source/AutoReviser.Tests/ImmutableExtensions_specs.cs

```diff
@@ -325,5 +359,45 @@ public class HasImmutableArrayProperty
 
             revision.Delta[1].Echo.Alfa.Should().Be(element);
         }
+
+        [TestMethod]
+        [AutoData]
+        public void Revise_updates_element_with_internal_constructor(
+            Guid alfa,
+            int bravo,
+            SimpleImmutableObject charlie,
+            SimpleImmutableObject expected)
+        {
+            // Arrange
+            var source = new ImmutableObjectWithInternalConstuctor(alfa, bravo, charlie);
+
+            // Act
+            var actual = source.Revise(x => x.Charlie == expected);
+
+            // Assert
+            actual.Alfa.Should().Be(alfa);
+            actual.Bravo.Should().Be(bravo);
+            actual.Charlie.Should().BeEquivalentTo(expected);
+        }
+
+        [TestMethod]
+        [AutoData]
+        public void Revise_updates_element_with_internal_properties(
```

Comment:
IMO this method tests that updating properties, not elements. So I recommend that you change the name of the test method.